datafusion/core/tests/sqllogictests/test_files/decimal.slt

```diff
@@ -0,0 +1,594 @@
+----
+1.00001
+1.00002
+1.00002
+1.00003
+1.00003
+1.00003
+1.00004
+1.00004
+1.00004
+1.00004
+1.00005
+1.00005
+1.00005
+1.00005
+1.00005
+
+
+# array decimal(10,6) + array decimal(12,7) => decimal(13,7)
+query ?
+select arrow_typeof(c1+c5) from decimal_simple limit 1;
+----
+Decimal128(13, 7)
+
+
+query R
+select c1+c5 from decimal_simple;
+----
+0.000024
+0.000045
+0.000039
+0.000062
+0.000065
+0.000041
+0.000084
+0.00008
+0.00008
+0.000084
+0.000102
+0.000128
+0.000083
+0.000118
+0.00015
+
+
+query ?
+select arrow_typeof(c1-1) from decimal_simple limit 1;
+----
+Decimal128(27, 6)
+
+
+query R
+select c1-1 from decimal_simple;
+----
+-0.99999
+-0.99998
+-0.99998
+-0.99997
+-0.99997
+-0.99997
+-0.99996
+-0.99996
+-0.99996
+-0.99996
+-0.99995
+-0.99995
+-0.99995
+-0.99995
+-0.99995
+
+
+query ?
+select arrow_typeof(c1-c5) from decimal_simple limit 1;
+----
+Decimal128(13, 7)
+
+
+query R
+select c1-c5 from decimal_simple;
+----
+-0.000004
+-0.000005
+0.000001
+-0.000002
+-0.000005
+0.000019
+-0.000004
+0
+0
+-0.000004
+-0.000002
+-0.000028
+0.000017
+-0.000018
+-0.00005
+
+
+query ?
+select arrow_typeof(c1*20) from decimal_simple limit 1;
+----
+Decimal128(31, 6)
+
+
+query R
+select c1*20 from decimal_simple;
+----
+0.0002
+0.0004
+0.0004
+0.0006
+0.0006
+0.0006
+0.0008
+0.0008
+0.0008
+0.0008
+0.001
+0.001
+0.001
+0.001
+0.001
+
+
+query ?
+select arrow_typeof(c1*c5) from decimal_simple limit 1;
+----
+Decimal128(23, 13)
+
+
+query R
+select c1*c5 from decimal_simple;
+----
+0.00000000014
+0.0000000005
+0.00000000038
+0.00000000096
+0.00000000105
+0.00000000033
+0.00000000176
+0.0000000016
+0.0000000016
+0.00000000176
+0.0000000026
+0.0000000039
+0.00000000165
+0.0000000034
+0.000000005
+
+
+query ?
+select arrow_typeof(c1/cast(0.00001 as decimal(5,5))) from decimal_simple limit 1;
+----
+Decimal128(21, 12)
+
+
+query R
+select c1/cast(0.00001 as decimal(5,5)) from decimal_simple;
+----
+1
+2
+2
+3
+3
+3
+4
+4
+4
+4
+5
+5
+5
+5
+5
+
+
+query ?
+select arrow_typeof(c1/c5) from decimal_simple limit 1;
+----
+Decimal128(30, 19)
+
+
+query R
+select c1/c5 from decimal_simple;
+----
+0.714285714286
+0.8
+1.052631578947
+0.9375
+0.857142857143
+2.727272727273
+0.909090909091
+1
+1
+0.909090909091
+0.961538461538
+0.641025641026
+1.515151515152
+0.735294117647
+0.5
+
+
+query ?
+select arrow_typeof(c5%cast(0.00001 as decimal(5,5))) from decimal_simple limit 1;
+----
+Decimal128(7, 7)
+
+
+query R
+select c5%cast(0.00001 as decimal(5,5)) from decimal_simple;
+----
+0.000004
+0.000005
+0.000009
+0.000002
+0.000005
+0.000001
+0.000004
+0
+0
+0.000004
+0.000002
+0.000008
+0.000003
+0.000008
+0
+
+
+query ?
+select arrow_typeof(c1%c5) from decimal_simple limit 1;
+----
+Decimal128(11, 7)
+
+
+query R
+select c1%c5 from decimal_simple;
+----
+0.00001
+0.00002
+0.000001
+0.00003
+0.00003
+0.000008
+0.00004
+0
+0
+0.00004
+0.00005
+0.00005
+0.000017
+0.00005
+0.00005
+
+
+query ?
+select arrow_typeof(abs(c1)) from decimal_simple limit 1;
+----
+Float64
```

Comment:
The original test didn't check the type of `abs`, but I added a test entry for it. It seems like a bug to me. `Decimal(10,6)` was converted to a `Float64` as a result of `abs` call. It is probably related to https://github.com/apache/arrow-datafusion/issues/1335.